### PR TITLE
API provides geolocation to show articles based on location

### PR DESCRIPTION
**Pull request by Odeane**
### User Story
```
As a newspaper API
In order for users to read relevant articles close to their location 
I would like to provide geolocation. 
```
## Changes proposed in this pull request:
* Added Geolocation

## What I have learned working on this feature:
* We learned how to use geocoder.

## Gems added
-  Geocoder

### CODE REVIEW

File: spec/models/article_spec.rb [hunk 1]
```diff
@@ -11,12 +11,17 @@
     it { is_expected.to have_db_column :lead }
     it { is_expected.to have_db_column :content }
     it { is_expected.to have_db_column :category }
+    it { is_expected.to have_db_column :journalist_id }
+    it { is_expected.to have_db_column :published }
+    it { is_expected.to have_db_column :location }
+
```
Comment (viktorsommar) [inline spec/models/article_spec.rb]: Remove space

File: spec/models/article_spec.rb [hunk 2]
```diff
@@ -11,12 +11,17 @@
...
+    it { is_expected.to have_db_column :location }
+    
   end
 
   describe 'Validations' do
     it { is_expected.to validate_presence_of :title }
     it { is_expected.to validate_presence_of :lead }
     it { is_expected.to validate_presence_of :content }
     it { is_expected.to validate_presence_of :category }
+
```
Comment (viktorsommar) [inline spec/models/article_spec.rb]: Remove space

File: spec/requests/api/v1/articles/local_news_spec.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+RSpec.describe "GET /v1/articles?longitude&latitude", type: :request do
+  let!(:swedish_articles) { 5.times { create(:article, location: 'Sweden') } }
+  let!(:danish_articles) { 5.times { create(:article, location: 'Denmark') } }
+
+  describe 'successfully' do
```
Comment (viktorsommar) [inline spec/requests/api/v1/articles/local_news_spec.rb:5]: Are you not adding a sad path?

**Review by viktorsommar — CHANGES_REQUESTED** [review]

File: spec/requests/api/v1/articles/local_news_spec.rb (hunk 3 above)
Comment (Odeane) [reply spec/requests/api/v1/articles/local_news_spec.rb:5]: we did.

File: app/models/article.rb [hunk 4]
```diff
@@ -1,6 +1,6 @@
 class Article < ApplicationRecord
   validates_presence_of :title, :lead, :content, :category
-  enum category: [:culture, :economy, :international, :lifestyle, :local, :sports]
+  enum category: [:culture, :economy, :international, :lifestyle, :sports]
```
Comment (viktorsommar) [inline app/models/article.rb:3]: Why did we remove local as a category?

File: spec/requests/api/v1/articles/show_spec.rb [hunk 5]
```diff
@@ -24,12 +24,12 @@
       get "/api/v1/articles/3"
     end
 
-  it 'responds with 422 status' do
-    expect(response).to have_http_status 422
-    end
+    it 'responds with 422 status' do
+      expect(response).to have_http_status 422
+      end
```
Comment (viktorsommar) [inline spec/requests/api/v1/articles/show_spec.rb]: indentation

File: spec/requests/api/v1/articles/show_spec.rb [hunk 6]
```diff
@@ -24,12 +24,12 @@
...
 
-  it 'responds with error message' do
-    expect(response_json["message"]).to eq 'Unfortunatly the article you were looking for could not be found.'
+    it 'responds with error message' do
+      expect(response_json["message"]).to eq 'Unfortunatly the article you were looking for could not be found.'
+      end
```
Comment (viktorsommar) [inline spec/requests/api/v1/articles/show_spec.rb]: indentation

**Review by viktorsommar — CHANGES_REQUESTED** [review]

File: app/models/article.rb (hunk 4 above)
Comment (Odeane) [reply app/models/article.rb:3]: because it doesn't correlate with what we are doing on the front end.

**Review by oliverochman — APPROVED** [review]

**Pull request merged**